src/context/ProfileContext.jsx

```diff
@@ -0,0 +1,44 @@
+import { createContext, useContext, useState } from 'react';
+import { useEffect } from 'react';
+import { Redirect } from 'react-router-dom';
+import { getProfile } from '../services/profile';
+
+export const ProfileContext = createContext();
+
+const ProfileProvider = ({ children }) => {
+  const [profile, setProfile] = useState({});
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const data = await getProfile();
+        setProfile(data);
+      } catch (error) {
+        <Redirect to="/profile/form" />;
```

Comment:
We should avoid having components inside of the useEffect, a history.push or replace would make more sense here